lib/semantics/check-return.cc

```diff
@@ -33,18 +30,16 @@ void ReturnStmtChecker::Leave(const parser::ReturnStmt &returnStmt) {
   // C1575 The scalar-int-expr is allowed only in the inclusive scope of a
   // subroutine subprogram.
   const auto &scope{context_.FindScope(context_.location().value())};
-  const auto *subprogramScope{FindContainingSubprogram(scope)};
-  if (!subprogramScope) {
-    context_.Say(
-        "RETURN must in the inclusive scope of a SUBPROGRAM"_err_en_US);
-    return;
-  }
-  if (returnStmt.v && subprogramScope->kind() == Scope::Kind::Subprogram) {
-    if (IsFunction(*subprogramScope->GetSymbol())) {
+  if (const auto *subprogramScope{FindContainingSubprogram(scope)}) {
+    if (returnStmt.v &&
+        (subprogramScope->kind() == Scope::Kind::MainProgram ||
```

Comment:
Could you add here also for Block Data the same message.
Is it possible to test  Scope::Kind::BlockData?
When I run this test
block data bdpointer
  common /test/ t1
  return t1
end block data
It shows the standard message that I did something wrong, but nothing specific as saying return only in a subroutine.

Reply:
A `BLOCK DATA` subprogram cannot contain any executable statements.  This restriction is implemented in the parser.